### PR TITLE
feat(lambda): Ability to get details of an individual function #4214

### PR DESCRIPTION
**Pull request by saimadineni**
Current implemenation of /getfunctions returns a list of all the lambda functions,
but it doesn't take any request parameters to get individual functions. 
I've implemented the feature by making some changes.

### CODE REVIEW

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java [hunk 1]
```diff
@@ -24,28 +24,44 @@
...
+                             @RequestParam(value="account",required = false) String account) {
+    if (functionName == null || functionName.isEmpty()) {
+      return functionProviders.stream()
+        .map(FunctionProvider::getAllFunctions)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
+      }
+    else {
+      try{
+        System.out.println(functionName + "printingfunctionNametwo");
```
Comment (edwinavalos) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java]: move this to debug logging, or someone kind of logging or remove it

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java [hunk 2]
```diff
@@ -24,28 +24,44 @@
...
+        List<Function> myFunction = functionProviders.stream().map(functionProvider -> functionProvider.getFunction(account, region, functionName)).collect(Collectors.toList());
+        return myFunction;
+      } catch (NotFoundException e) {
+        throw new NotFoundException(functionName + "does not exist");
+      }
+    }
+    }
   }
-}
+
```
Comment (edwinavalos) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java]: Just one extra new line here

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java [hunk 3]
```diff
@@ -24,28 +24,44 @@
...
-
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Optional;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.apache.log4j.Logger;
+import java.util.*;
```
Comment (edwinavalos) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java]: I think the star imports are highly frowned upon.

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java (hunk 2 above)
Comment (saimadineni) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/FunctionController.java]: Took care of ti

**Comment by saimadineni** [comment]
I'm having issues squashing/rebasing my commits. Will create a new branch and update it.

**Pull request closed without merging**